addons/website_event/controllers/main.py

```diff
@@ -245,16 +252,70 @@ def _process_tickets_form(self, event, form_details):
+        else display the attendee registration modal.
+
+        For the slot registration modal, retrieving all the slots with available seats
+        for the ticket selection and pass them to the slot registration modal.
+        If there is none, showing an availability check error in the modal.
+        Only one slot can be selected for the current registration.
+        The attendee registration form is displayed after the slot selection modal.
+        """
         tickets = self._process_tickets_form(event, post)
         availability_check = True
-        if event.seats_limited:
-            ordered_seats = 0
-            for ticket in tickets:
-                ordered_seats += ticket['quantity']
-            if event.seats_available < ordered_seats:
-                availability_check = False
+        available_slots = self.env['event.slot']
+        event_slots = event.sudo().slot_ids
+        if not event.seats_limited:
+            available_slots = event_slots
+        else:
+            if event.is_multi_slots and any(not ticket.is_expired for ticket in event.no_slot_ticket_ids):
+                # multi slots with tickets check
+                ordered_seats_per_ticket = {ticket_data['ticket'].id: ticket_data['quantity'] for ticket_data in tickets}
+                for parent_ticket, slot_tickets in event.event_ticket_ids.grouped('parent_ticket_id').items():
+                    ordered_quantity = ordered_seats_per_ticket.get(parent_ticket.id)
+                    if not ordered_quantity:
+                        continue
+                    available_slots += slot_tickets.filtered(lambda slot_ticket: slot_ticket.seats_available >= ordered_quantity).mapped('slot_id')
+                if not available_slots:
+                    availability_check = False
+            else:
+                # not multi slots / multi slots without tickets check
+                ordered_seats = 0
+                for ticket in tickets:
+                    ordered_seats += ticket['quantity']
+                seats_available = max(slot.seats_available for slot in event_slots) if event.is_multi_slots else event.seats_available
+                if seats_available < ordered_seats:
+                    availability_check = False
+                if event.is_multi_slots:
+                    available_slots = event_slots.filtered(lambda slot: slot.seats_available >= ordered_seats)
         if not tickets:
             return False
+        if event.is_multi_slots:
+            return request.env['ir.ui.view']._render_template("website_event.modal_slot_registration", {
+                'tickets': tickets,
+                'event': event,
+                'availability_check': availability_check,
+                'available_slots_per_date': available_slots.sorted('date').grouped('date'),
+                'selected_tickets': post,
+            })
+        return request.env['ir.ui.view']._render_template("website_event.registration_attendee_details", {
```

Comment:
Would there be a simple way to avoid 'piling up' modals ? (the background gets darker each time) no need to rewrite everything, just asking